### DISCUSSION TITLE
What is the necessary and sufficient list of dirs need to be volumed for the db instance to be fully persistent?

### DISCUSSION THREAD

**Discussion by catostrophe** [gvenzl/oci-oracle-xe#253] [Q&A] [answered]
The container should correctly survive restarts.

What are exact locations of data files, control files, redo logs, alert logs, traces, archive logs, etc.?

**Comment by gvenzl** [comment]
Hi @catostrophe,

Thanks a lot for using these images!

All database-related files are in `/opt/oracle/oradata`
Archive log mode is not being used in these images.
Trace files can be found in `/opt/oracle/diag`

Hope this helps!

**Reply by catostrophe** [reply]
Thanks. I already use them as volumes. I also thought about adding spfile location in case I need to do ALTER SYSTEM. WDYT?

**Reply by gvenzl** [reply, marked as answer]
The spfile is already persisted in a subfolder called `dbconfig/<ORACLE_SID>`

**Comment by euasier** [comment]
I'm having the same problem, the database doesn't survive the container removal. Could you please specify which volumes to map ? Thanks

**Reply by euasier** [reply]
Solved, it is documented for your image as:
"Run a new persistent database container (data is kept throughout container lifecycles):

docker run -d -p 1521:1521 -e ORACLE_PASSWORD=<your password> -v oracle-volume:/opt/oracle/oradata gvenzl/oracle-xe
"
